server/lib/nl/constants.py

```diff
@@ -155,10 +155,75 @@
 
 # TODO: remove this special casing when a better NER model is identified which
 # can always detect these.
-SPECIAL_PLACES: Set[str] = {'palo alto', 'mountain view'}
+SPECIAL_PLACES: Set[str] = {
+    'palo alto', 'mountain view', 'world', 'earth', 'african', 'asian',
+    'european', 'north american', 'south american', 'latin american', 'oceania',
+    'american'
+}
 
 SPECIAL_PLACE_REPLACEMENTS: Dict[str, str] = {'us': 'United States'}
 
+SPECIAL_PLACES_TO_GEOIDS: Dict[str, str] = {
+    # Earth
+    'earth': 'Earth',
+    'world': 'Earth',
+
+    # US
+    'american': 'country/USA',
+
+    # Continents
+    'africa': 'africa',
+    'antarctica': 'antarctica',
+    'asia': 'asia',
+    'europe': 'europe',
+    'north america': 'northamerica',
+    'northamerica': 'northamerica',
+    'south america': 'southamerica',
+    'southamerica': 'southamerica',
+    'oceania': 'oceania',
+
+    # Alternates
+    'african': 'africa',
+    'asian': 'asia',
+    'european': 'europe',
+    'north american': 'northamerica',
+    'northamerican': 'northamerica',
+    'south american': 'southamerica',
+    'southamerican': 'southamerica',
+    'african continent': 'africa',
+    'asian continent': 'asia',
+    'european continent': 'europe',
+    'north american continent': 'northamerica',
+    'south american continent': 'southamerica',
+    'latin america': 'southamerica',
+    'latinamerica': 'southamerica',
+    'latin american': 'southamerica',
+    'latinamerican': 'southamerica',
+    'latin american continent': 'southamerica',
+    'australian continent': 'oceania',
+    'australasia': 'oceania',
+}
+
+MAPS_API = "https://maps.googleapis.com/maps/api/place/textsearch/json?"
+
+MAPS_GEO_TYPES = {
```

Comment:
my py is bad... is this is a list?  Puzzled by `{}`

Reply:
using frozenset() now